builtin/logical/pki/chain_test.go

```diff
@@ -525,14 +566,37 @@ func (c CBIssueLeaf) Run(t testing.TB, b *backend, s logical.Storage, knownKeys
 	raw_crl := resp.Data["crl"].(string)
 	crl := ToCRL(t, raw_crl, issuer)
 
-	foundCert := false
-	for _, revoked := range crl.TBSCertList.RevokedCertificates {
-		// t.Logf("[%v] revoked serial number: %v -- vs -- %v", index, revoked.SerialNumber, cert.SerialNumber)
-		if revoked.SerialNumber.Cmp(cert.SerialNumber) == 0 {
-			// t.Logf("found revoked cert at index: %v", index)
-			foundCert = true
-			break
+	foundCert := requireSerialNumberInCRL(nil, crl.TBSCertList, api_serial)
+	if !foundCert {
+		if !hasCRL && !isDefault {
+			// Update the issuer we expect to find this on.
+			resp, err := CBRead(b, s, "config/issuers")
+			if err != nil {
+				t.Fatalf("failed to read default issuer config: %v", err)
+			}
+			if resp == nil {
+				t.Fatalf("failed to read default issuer config: nil response")
+			}
+			defaultID := resp.Data["default"].(issuerID).String()
+			c.Issuer = defaultID
+			issuer = nil
+			CBRead(b, s, "crl/rotate")
```

Comment:
I don't understand why the crl/rotate call would need to be done here? The revoke should have forced the cert to be on an expected CRL?

Reply:
Ah that's right. I had a bug earlier in my tests where I forgot about duplicate issuers, so I had added this call and we don't actually need it.